mysql-test/t/acl_mdl_scalability.test

```diff
@@ -0,0 +1,255 @@
+--source include/count_sessions.inc
+--source include/have_debug.inc
+--source include/have_debug_sync.inc
+# We rely on g++ to compile c-program
+--source include/not_windows.inc
+# Because of 6K connections, mysqld consumes close to
+# 14G virtual memory
+--source include/big_test.inc
+
+--echo # Check if system can handle 12k concurrent threads
+--echo # 12K because server creates 6K threads and clients
+--echo # another 6k threads/processes
+
+--echo # There is no easy way to check if the system can
+--echo # handle 12K threads. We compile C-program (need g++)
+--echo # and execute it and check return status
+
+--write_file $MYSQL_TMP_DIR/thread_limit_test.cc
+#include <pthread.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <string.h>
+#include <atomic>
+
+#define NUM_THREADS     12000
+std::atomic<int> current_threads{0};
+
+void* ThreadTask(void *threadid)
+{
+   long tid;
+   tid = (long)threadid;
+   printf("Hello World! It's me, thread #%ld!\n", tid);
+   std::atomic_fetch_add(&current_threads, 1);
+   while (current_threads.load() < NUM_THREADS) {
+           //printf("sleeping in thread id #%ld!\n", tid);
+           sleep(1);
+   }
+   printf("Thread exiting #%ld\n", tid);
+   pthread_exit(NULL);
+}
+
+int main ()
+{
+   pthread_t threads[NUM_THREADS];
+   int rc;
+   for(long t=0; t<NUM_THREADS; t++){
+      printf("In main: creating thread %ld\n", t);
+      rc = pthread_create(&threads[t], NULL, ThreadTask, (void *)t);
+      if (rc){
+         printf("ERROR; return code from pthread_create() is %s\n", strerror(rc));
+         exit(-1);
```

Comment:
Under ASan, because of resource leaks here, you will get a different exit code (`42`). Please, double check if this is OK.

Reply:
As long as it is not zero, we are OK

Reply:
What prevents us from stopping/joining previously created threads here to avoid the leak?

Reply:
we can.. since this is a test program for testcase and since we terminate, they will get cleaned up.